README.md

```diff
@@ -7388,105 +7387,128 @@ See [SQLCipher 4.0.0 Release](https://www.zetetic.net/blog/2018/11/30/sqlcipher-
 
 **You can change the passphrase** of an already encrypted database.
 
-The technique to do so is not the same, depending on whether you are using a [database queue](#database-queues) or a [database pool](#database-pools).
-
-For database queues, first open the database with the old passphrase, and then apply the new passphrase:
+When you use a [database queue](#database-queues), open the database with the old passphrase, and then apply the new passphrase:
 
 ```swift
-// 1. Open the database queue
-var configuration = Configuration()
-configuration.prepareDatabase = { db in
-    try db.usePassphrase("oldSecret")
-}
-let dbQueue = try DatabaseQueue(path: "...", configuration: configuration)
-
-// 2. Later on, change the passphrase
 try dbQueue.write { db in
     try db.changePassphrase("newSecret")
 }
+```
 
-// 3. Keep using the database queue with the new passphrase
-try dbQueue.write { ... }
-try dbQueue.read { ... }
+When you use a [database pool](#database-pools), make sure that no concurrent read can happen during the passphrase change with the `barrierWriteWithoutTransaction` method, and that all future reads open a new database connection with the `invalidateReadOnlyConnections` method:
+
+```swift
+try dbPool.barrierWriteWithoutTransaction { db in
+    try db.changePassphrase("newSecret")
+    dbPool.invalidateReadOnlyConnections()
+}
 ```
 
-For database pools, it is a little bit more complicated. This is because pools create SQLite connections on demand. Your application must provide a storage for the passphrase, ready to be queried whenever a new SQLite connection should be created. This "storage" can be, for example, the system keychain.
+> :point_up: **Note**: When an application wants to keep on using a database queue or pool after the passphrase has changed, it is responsible for providing the correct passphrase to the `usePassphrase` method called in the database preparation function.
```

Comment:
I'm not totally clear on what this note is trying to prevent. Is it cases like this:

```
// this won't work across a password change
let passphrase = getStoredPassphrase()
prepareDatabase { db 
    try db.usePassphrase(passphrase)
}
```

```
// This could work across a password change
prepareDatabase { db 
    let passphrase = getStoredPassphrase()
    try db.usePassphrase(passphrase)
}
```

Reply:
Yes, this is it. Do you think we should provide guidance here, as below?

> :point_up: **Note**: When an application wants to keep on using a database queue or pool after the passphrase has changed, it is responsible for providing the correct passphrase to the `usePassphrase` method called in the database preparation function. Consider:
>
> ```swift
> // WRONG: this won't work across a passphrase change
> let passphrase = try getPassphrase()
> var config = Configuration()
> config.prepareDatabase = { db in
>     try db.usePassphrase(passphrase)
> }
>
> // CORRECT: get the latest passphrase when it is needed
> var config = Configuration()
> config.prepareDatabase = { db in
>     let passphrase = try getPassphrase()
>     try db.usePassphrase(passphrase)
> }
> ```

Reply:
OK let's do it: a1c96f909333e81909e1cfd06b335c42cc1147ab

Reply:
Yes I think that's better. 👍